MuseumsOfChicago/GeoBlazor.Sample.Museums/Components/Pages/Home.razor

```diff
@@ -0,0 +1,249 @@
+                            {nameof(MuseumRecord.Website), record.Website}
+                        });
+
+            Graphic graphic = new Graphic(point, museumSymbol, popupTemplate, attributes);
+            graphics.Add(graphic);
+        }
+
+        await _museumLayer.Add(graphics);
+    }
+
+
+    private async Task OnFilterChanged(ChangeEventArgs arg)
+    {
+        _filterValue = arg.Value!.ToString();
+        await LoadMuseums();
+    }
+
+    private List<MuseumRecord> FilteredRecords =>
+        string.IsNullOrWhiteSpace(_filterValue)
+            ? _records
+            : _records.Where(r => r.Discipline.Equals(_filterValue)).ToList();
+
+
+    private async Task ToggleRouting()
+    {
+        _routing = !_routing;
+        if (!_routing)
+        {
+            await Clear();
+        }
+        await _popupWidget!.SetProperty(nameof(PopupWidget.AutoOpenEnabled), !_routing);
+    }
+
+    private async Task OnClick(ClickEvent clickEvent)
+    {
+        if (!_routing) return;
+        Graphic stop = new(clickEvent.MapPoint);
+        await _routeLayer!.Add(stop);
+        List<Graphic> stops = _routeParameters.FeatureSetStops!.Features!.ToList();
+        stops.Add(stop);
+        _routeParameters.FeatureSetStops.Features = stops.ToArray();
+
+        if (_routeParameters.FeatureSetStops.Features.Length >= 2)
+        {
+            RouteSolveResult result = await Route.Solve(_routeUrl, _routeParameters);
+            await ShowRoute(result);
+        }
+    }
+
+    private async Task ShowRoute(RouteSolveResult result)
+    {
+        Graphic? routeGraphic = result.RouteResults?.FirstOrDefault()?.Route;
+
+        if (routeGraphic is null) return;
+        await routeGraphic.SetSymbol(_routeSymbol);
+        await _routeLayer!.Add(routeGraphic);
+        _directions = result.RouteResults!.Select(r => r.Directions!).ToArray();
+    }
+
+    private async Task Clear()
+    {
+        await _popupWidget!.Close();
+        _directions = [];
+        _routeParameters.FeatureSetStops!.Features = [];
+        await _routeLayer!.Clear();
+    }
+
+    private readonly SimpleLineSymbol _routeSymbol = new(new MapColor(0, 0, 255, 0.5), 5);
+    private readonly RouteParameters _routeParameters = new()
+    {
+        FeatureSetStops = new FeatureSet
+        {
+            Features = []
+        },
+        ReturnDirections = true
+    };
+    private readonly string _routeUrl = "https://route-api.arcgis.com/arcgis/rest/services/World/Route/NAServer/Route_World";
+    private DirectionsFeatureSet[] _directions = [];
+
+
+    #region Private Fields
+    private MapView? _mapView;
+    private GraphicsLayer? _museumLayer;
+    private PopupWidget? _popupWidget;
+    private GraphicsLayer? _routeLayer;
+    private MapImageLayer? _trafficLayer;
+
+    private List<MuseumRecord> _records = [];
+    private bool _isLoaded;
+    private bool _showTraffic;
+    private bool _routing;
```

Comment:
I'm thinking this name could be refined a bit.  As it is, I could see it meaning whether a routing operation is in-progress or whether routing functionality is enabled.